central/notifications/datastore/datastore_impl.go

```diff
@@ -0,0 +1,64 @@
+package datastore
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/notifications/datastore/internal/search"
+	"github.com/stackrox/rox/central/notifications/datastore/internal/store"
+	"github.com/stackrox/rox/central/notifications/datastore/internal/writer"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/errox"
+)
+
+var _ DataStore = (*datastoreImpl)(nil)
+
+type datastoreImpl struct {
+	searcher search.Searcher
+	store    store.Store
+	// Buffered writer. Flush to initiate batched upsert.
+	writer   writer.Writer
+}
+
+func (ds *datastoreImpl) AddNotification(ctx context.Context, notification *storage.Notification) error {
+	if err := ds.writer.Upsert(ctx, notification); err != nil {
```

Comment:
Since this is something atypical for our datastores, let's leave a comment that the writer implementation will handle SAC checks (I've added a comment on the other PR calling out that this is currently missing).